agent/structs/config_entry.go

```diff
@@ -362,6 +363,10 @@ func (e *ProxyConfigEntry) Normalize() error {
 	}
 
 	e.Kind = ProxyDefaults
+
+	if e.Name != "" && e.Name != ProxyConfigGlobal {
+		return fmt.Errorf("invalid name (%q), only %q is supported", e.Name, ProxyConfigGlobal)
+	}
```

Comment:
I would add a comment here to describe why this needed to be here instead of `Validate()`.